### PR TITLE
feat(app): Add basic UI for swapping a pipette

### PR DESCRIPTION
**Pull request by mcous**
## overview

This PR adds enough UI to actually swap a pipette out!

- Ticks off the "warn that removing a pipette will clear related calibration data" checkbox in #860
- Addresses all of #863 

...GIF to be inserted here when I get on a robot...

## changelog

- feat(app): Add basic UI for swapping a pipette 
    - Includes using new pipette definitions in `@opentrons/labware-definitions`

## review requests

Robot testing and standard code review

### TODO in followup PR(s)

- App does not yet disable plunger motor (last remaining checkbox in #860)
- Screens have been updated while this work was in progress, so they haven't been implemented yet:
    - Updated modal styles
    - Updated error messages for failure to detach and attaching the wrong pipette
    - Back button saying "Exit" if clicking it will exit
    - This will probably be easiest if we tack on one more ticket for these items
- All of #861

### CODE REVIEW

**Comment by IanLondon** [comment]
Tried to test on the QA bot b/c others were taken -- server 3.0.0-beta.6. I'm not sure what pipette is on there, but it didn't detect it so I couldn't get through. Also app hit a 404 when it called `home/` in response to cancelling/exiting the change pipette modal. Will test later on another bot.

File: app/src/robot/types.js [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 export type State = $Call<reducer>
 
 // TODO Ian 2018-02-27 files that import from here should just import from @opentrons/components directly
-export type {
-  Mount,
-  Channels
-}
+export type {Mount}
+
+export type Channels = PipetteChannels
```
Comment (IanLondon) [inline app/src/robot/types.js]: Maybe we should have a TODO to remove this import/export rerouting eventually?

File: app/src/components/ChangePipette/PipetteSelection.js [hunk 2]
```diff
@@ -1,20 +1,28 @@
...
 import styles from './styles.css'
 
 const LABEL = 'Select the pipette you wish to attach:'
 
-export type PipetteSelectionProps = React.ElementProps<typeof DropdownField>
+export type PipetteSelectionProps = {
+  onChange: $PropertyType<React.ElementProps<typeof DropdownField>, 'onChange'>
+}
+
+const OPTIONS = getPipetteModels().map(m => {
```
Comment (IanLondon) [inline app/src/components/ChangePipette/PipetteSelection.js]: `PIPETTE_OPTIONS` might be universal enough to move into pipette defs

File: app/src/components/ChangePipette/index.js [hunk 3]
```diff
@@ -41,7 +38,7 @@ const TITLE = 'Pipette Setup'
 // used to guarentee mount param in route is left or right
 const RE_MOUNT = '(left|right)'
 // used to guarentee model param in route is a pipettes model
-const RE_MODEL = `(${PIPETTES.map(p => p.model).join('|')})`
+const RE_MODEL = `(${getPipetteModels().join('|')})`
```
Comment (IanLondon) [inline app/src/components/ChangePipette/index.js]: What's `RE_` stand for?

File: app/src/robot/types.js (hunk 1 above)
Comment (mcous) [reply app/src/robot/types.js]: Ah yeah I guess the newline makes this a little ambiguous. In my head while writing your TODO above still applied to this line

File: app/src/components/ChangePipette/index.js (hunk 3 above)
Comment (mcous) [reply app/src/components/ChangePipette/index.js]: Regex

File: app/src/robot/types.js (hunk 1 above)
Comment (IanLondon) [reply app/src/robot/types.js]: d'oh yeah I can't read

**Comment by pantslakz** [comment]
@Kadee80 there's designs for that in Zeplin. Would definitely be helpful to put this in - otherwise users will have no way of knowing if they have to switch pipettes.

screen shot 2018-04-23 at 5 46 38 pm

@mcous apologies as I'm still unable to review this. hoping to connect with Andy today.

**Comment by mcous** [comment]
Walked through this with @umbhau on Sunset bot and got the 👍 for merging. Will follow up on items in the TODO above for the rest of the day. @pantslakz if you're set up by tomorrow you should be able to test the more complete experience

**Pull request merged**